### PR TITLE
Common strs

### PR DESCRIPTION
**Pull request by olliethomas**
This is not a very exciting PR. It just adds two enums to avoid potentially mis-typing common (albeit short) words.

On the housekeeping side, ran `poetry update`, and add `pre-commit` to the dev dependencies (doesn't really fit into an existing group).

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -73,6 +73,10 @@ optional = true
 [tool.poetry.group.torchcpu.dependencies]
 torch = { version = "*", source = "torchcpu", markers = "sys_platform == 'linux'" }
 
+
+[tool.poetry.group.dev.dependencies]
+pre-commit = "^3.4.0"
+
```
Comment (tmke8) [inline pyproject.toml:79]: Yup, seems like a good group to put it in :+1:

**Review by tmke8 — APPROVED** [review]

File: docs/conf.py [hunk 2]
```diff
@@ -10,12 +10,12 @@
...
 # documentation root, use os.path.abspath to make it absolute, like shown here.
 #
-import os
+from pathlib import Path
 import sys
 
 import toml
 
-sys.path.insert(0, os.path.abspath(".."))
+sys.path.insert(0, str(Path("..").resolve()))
```
Comment (olliethomas) [inline docs/conf.py:18]: ruff insisted that this should be changed and made this suggestion. I have not tested it.

Comment (tmke8) [reply docs/conf.py:18]: seems fine

File: tests/torch_data_test.py [hunk 3]
```diff
@@ -17,10 +17,10 @@ def dummy_ds() -> TensorDataset:
 @pytest.mark.parametrize("as_indices", [False, True])
 @pytest.mark.parametrize("props", [0.5, [-0.2, 0.5], [0.1, 0.3, 0.4], [0.5, 0.6]])
 def test_prop_random_split(
-    dummy_ds: TensorDataset, props: float | list[float], as_indices: bool
+    dummy_ds: TensorDataset, props: float | int | list[float] | list[int], as_indices: bool
 ) -> None:
-    sum_ = props if isinstance(props, float) else sum(props)
-    props_ls = [props] if isinstance(props, float) else props
+    sum_ = props if isinstance(props, (float, int)) else sum(props)
+    props_ls = [props] if isinstance(props, (float, int)) else props
```
Comment (olliethomas) [inline tests/torch_data_test.py:23]: there must be a better way to solve this?

File: tests/torch_data_test.py [hunk 4]
```diff
@@ -17,10 +17,10 @@ def dummy_ds() -> TensorDataset:
 @pytest.mark.parametrize("as_indices", [False, True])
 @pytest.mark.parametrize("props", [0.5, [-0.2, 0.5], [0.1, 0.3, 0.4], [0.5, 0.6]])
 def test_prop_random_split(
-    dummy_ds: TensorDataset, props: float | list[float], as_indices: bool
+    dummy_ds: TensorDataset, props: float | int | list[float] | list[int], as_indices: bool
```
Comment (tmke8) [inline tests/torch_data_test.py:20]: The additional `int` is unnecessary; not sure about the `list[int]`

File: tests/torch_data_test.py (hunk 3 above)
Comment (tmke8) [reply tests/torch_data_test.py]: perhaps in python 3.10

**Pull request merged**